File: frontend/vue-vben-admin/apps/web-antd/src/api/system/alert.ts
import { requestClient } from '#/api/request';

export interface AlertItem {
  acknowledgeRemark?: null | string;
  acknowledgedAt?: null | string;
  acknowledgedBy?: null | string;
  content: string;
  firstTriggeredAt: string;
  id: string;
  lastTriggeredAt: string;
  level: string;
  recoveredAt?: null | string;
  source: string;
  status: string;
  title: string;
  triggerCount: number;
  type: string;
}

export interface AlertListParams {
  level?: string;
  page?: number;
  pageSize?: number;
  status?: string;
  type?: string;
}

export interface AlertListResult {
  items: AlertItem[];
  total: number;
}

export interface AcknowledgeAlertParams {
  remark?: null | string;
}

export async function getAlertListApi(params: AlertListParams) {
  return requestClient.get<AlertListResult>('/system/alert/list', {
    params,
  });
}

export async function acknowledgeAlertApi(
  id: string,
  data: AcknowledgeAlertParams,
) {
  return requestClient.post<AlertItem>(
    `/system/alert/${id}/acknowledge`,
    data,
  );
}

File: src/MiniAdmin.Domain/Entities/Alert.cs
namespace MiniAdmin.Domain.Entities;

public sealed class Alert
{
    public Guid Id { get; set; }

    public string Type { get; set; } = string.Empty;

    public string Level { get; set; } = string.Empty;

    public string Title { get; set; } = string.Empty;

    public string Content { get; set; } = string.Empty;

    public string Source { get; set; } = string.Empty;

    public string Status { get; set; } = "Active";

    public DateTimeOffset FirstTriggeredAt { get; set; } = DateTimeOffset.UtcNow;

    public DateTimeOffset LastTriggeredAt { get; set; } = DateTimeOffset.UtcNow;

    public DateTimeOffset? RecoveredAt { get; set; }

    public string? AcknowledgedBy { get; set; }

    public DateTimeOffset? AcknowledgedAt { get; set; }

    public string? AcknowledgeRemark { get; set; }

    public int TriggerCount { get; set; } = 1;

    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;

    public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.UtcNow;
}
